lib/logstash/inputs/kafka.rb

```diff
@@ -278,6 +285,13 @@ def thread_runner(logstash_queue, consumer)
           for record in records do
             codec_instance.decode(record.value.to_s) do |event|
               decorate(event)
+              if schema_registry_url
+                json = LogStash::Json.load(record.value.to_s)
```

Comment:
Do I understand that the value of the Kafka message that was in Avro has been converted in a json string and is again being parsed from json add the elements in the 'Logstash event object'?
Is it not possible to bypass the json and go from Kafka to 'Logstash event object' directly?
I am not sure how expensive the json parsing is.

Reply:
The record is an instance of [ConsumerRecord](https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/consumer/ConsumerRecord.html#ConsumerRecord-java.lang.String-int-long-K-V-) and could be potentially deeply nested. Here the desire is to copy all the keys and values from the record structure into the Logstash event. This is a simple way to do it, but if you have any evidence of bottleneck please raise an issue so that we can try another implementaion